lib/plugins/acts_as_activity_provider/lib/acts_as_activity_provider.rb

```diff
@@ -191,13 +191,13 @@ def restrict_projects_by_user(options, user, query)
               allowed_projects = []
 
               user.projects_by_role.each do |role, projects|
-                allowed_projects << projects.map(&:id) if role.allowed_to?(perm.name)
+                allowed_projects << projects.map(&:id) if perm && role.allowed_to?(perm.name)
```

Comment:
I concede. Seeing as the specs a green I trust that this change only fixes and does not break things.